WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderTagActions.java

```diff
@@ -102,51 +106,63 @@ public static boolean addTags(@NotNull final List<ReaderTag> tags,
                     ReaderTagType.FOLLOWED);
             newTags.add(newTag);
         }
-
-        com.wordpress.rest.RestRequest.Listener listener = jsonObject -> {
-            AppLog.i(T.READER, "add tag succeeded");
-            // the response will contain the list of the user's followed tags
-            ReaderTagList followedTags = parseFollowedTags(jsonObject);
-            ReaderTagTable.replaceFollowedTags(followedTags);
-            if (actionListener != null) {
-                ReaderActions.callActionListener(actionListener, true);
+        if (!isLoggedIn && AppPrefs.isReaderImprovementsPhase2Enabled()) {
+            // TODO: Revisit logic when ReaderTagRepository's getUserTags is implemented
+            if (!AppPrefs.getReaderRecommendedTagsDeletedForLoggedOutUser()) {
```

Comment:
I'm still a bit confused if we actually need to delete the "recommended" tags. However, I'd consider adding a comment either here or above the method declartion - explaining the history. Moreover, I'd consider explaining the history even on the PR so it's easier to find the answer why this code was ever added. Wdyt?

Reply:
So I think I have a better understanding of the code now. I'm still wondering if we could remove this whole if statement and always invoke `ReaderTagTable.replaceFollowedTags(newTags);`. This wouldn't delete the posts related to tags which are not in the database anymore - however, these would get deleted automatically [here](https://github.com/wordpress-mobile/WordPress-Android/blob/ef7bfa7c8ca6362c3890f4242b897ed29997cd57/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L208).
I might be missing something very important, so please don't assume it'll work :D. I just thing it's worth asking/exploring as it'd simplify the code. Let me know what you think :). I'll be happy to help exploring in case you can't foresee any issues from the top of your head.

Reply:
> always invoke ReaderTagTable.replaceFollowedTags(newTags);

This is what [replaceFollowedTags fn](https://github.com/wordpress-mobile/WordPress-Android/blob/87c1f4bf0a1297fdcf412042fd6181beb9cafc6b/WordPress/src/main/java/org/wordpress/android/datasets/ReaderTagTable.java#L97-L99) does
```
// first delete all existing followed tags, then insert the passed ones
String[] args = {Integer.toString(ReaderTagType.FOLLOWED.toInt())};
db.execSQL("DELETE FROM tbl_tags WHERE tag_type=?", args);
addOrUpdateTags(tags);
```

Will it not delete all locally saved followed tags for a logged out user? We're supposed to preserve them.

Reply:
Ohh, I see ... We want to delete the tags only the first time (that's the reason we can use "replaceFollowedTags" in this scenario). However, we don't want to delete it after. Thanks for clarification!